src/meili/map_matcher.cc

```diff
@@ -202,88 +201,82 @@ MatchResult FindMatchResult(const MapMatcher& mapmatcher,
     return CreateMatchResult(measurement);
   }
 
+  // Find the last edge of the path from the previous state to the current state
   const auto& state = mapmatcher.state_container().state(stateid);
-  // Construct the route from previous state to current state, and
-  // find out which edge it matches
-  baldr::GraphId prev_end;
-  baldr::GraphId prev_state_edgeid;
+  baldr::GraphId prev_edge;
   if (prev_stateid.IsValid()) {
     const auto& prev_state = mapmatcher.state_container().state(prev_stateid);
     // It must stay on the last edge of the route
     const auto rbegin = prev_state.RouteBegin(state), rend = prev_state.RouteEnd();
     if (rbegin != rend && rbegin->edgeid().Is_Valid()) {
-      prev_state_edgeid = rbegin->edgeid();
-      prev_end = graph_reader->directededge(prev_state_edgeid)->endnode();
+      prev_edge = rbegin->edgeid();
     }
   }
 
-  // Do the same from current state to next state
-  baldr::GraphId next_start;
-  baldr::GraphId next_state_edgeid;
+  // Find the first edge of the path from the current state to the next state
+  baldr::GraphId next_edge;
   if (next_stateid.IsValid()) {
```

Comment:
we could actually skip getting the next_edge if we got a previous one except in the case where we need to check for intersection based routing below. perhaps its worth figuing out how to put this code below so it only happens in that case..